### PR TITLE
man fix tableColFormat digit on the end

### PR DESCRIPTION
**Pull request by Yanpas**
relates to #5026 
Fixes cases like:
```
3-a/man3p//basename.3p

Error at "source" (line 105, column 3):
unexpected "5"
expecting lf new-line, ",", white space or "."
lf5 | lf5.
```

### CODE REVIEW

**Comment by Yanpas** [comment]
After 1st commit all posix files are not parsed due to nf macros. But one file failed to parse on this line:
```
Error at "source" (line 27, column 14):
unexpected "("
expecting lf new-line, ",", white space or "."
lw(15) | cf5w(15) | lw(40).
             ^
Failed to convert /mnt/d/dev/mandocset/resource/man-pages-posix-2013-a/man0p//tar.h.0p
```

File: src/Text/Pandoc/Readers/Roff.hs [hunk 1]
```diff
@@ -433,12 +433,13 @@ tableColFormat = do
                 else return ""
       return $ x : num
     pipeSuffix' <- option False $ True <$ string "|"
+    skipMany (letter <|> digit)
```
Comment (jgm) [inline src/Text/Pandoc/Readers/Roff.hs]: I don't think we should skip these at this (tokenization) phase.

Comment (jgm) [reply src/Text/Pandoc/Readers/Roff.hs]: Perhaps the better solution is to move line 424 (numsuffixes) to right before 434.  That is, the numerical suffix comes after all other suffixes.  I wasn't clear from the tbl manual whether it worked this way, but it seems from your examples that maybe it does...

**Comment by jgm** [comment]
So far we're only parsing "plain" tables with inline-level cell contents, so we fail on a table containing `.nf`. But I've fixed things so we get a regular parse failure (falling back to TABLE and a message about skipping the content) rather than a parse error.

**Comment by jgm** [comment]
It would be easy to allow block-level content in cells, but then we'd need to have widths in addition to alignments.  Widths could be derived from the format spec or we could just set to equal widths for now.

**Comment by jgm** [comment]
Looks like a digit can occur in any order.

**Comment by jgm** [comment]
It was easy to fix this with these examples, thanks!

**Pull request closed without merging**